idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -26,32 +27,245 @@
+                                   b.params.true_species_set,
+                                   rule=rule_alpha_expr,
+                                   doc="Non-randomness parameters"))
+
+        # Calculate G terms
+        def rule_G_expr(b, i, j):
+            Y = getattr(b, pname+"_Y")
+
+            def _G_appr(b, pobj, i, j, T):
+                return exp(-alpha_rule(b, pobj, i, j, T) *
+                           tau_rule(b, pobj, i, j, T))
+
+            if ((pname, i) not in b.params.true_phase_component_set or
+                    (pname, j) not in b.params.true_phase_component_set):
+                return Expression.Skip
+            elif (i == j or
+                  (i in b.params.cation_set and j in b.params.cation_set) or
+                  (i in b.params.anion_set and j in b.params.anion_set)):
+                # No like species interactions
+                return Expression.Skip
+            elif ((i in molecular_set) and
+                    (j in molecular_set)):
+                return _G_appr(b, pobj, i, j, b.temperature)
+            elif (i in b.params.cation_set and j in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (i+", "+k), j,  b.temperature)
+                    for k in b.params.anion_set)
+            elif (j in b.params.cation_set and i in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (j+", "+k), i, b.temperature)
+                    for k in b.params.anion_set)
+            elif (i in b.params.anion_set and j in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (k+", "+i), j, b.temperature)
+                    for k in b.params.cation_set)
+            elif (j in b.params.anion_set and i in molecular_set):
+                return sum(
+                    Y[k] * _G_appr(b, pobj, (k+", "+j), i, b.temperature)
+                    for k in b.params.cation_set)
+            elif (i in b.params.cation_set and j in b.params.anion_set):
+                return sum(Y[k] * _G_appr(
+                    b, pobj, (i+", "+j), (k+", "+j), b.temperature)
+                    for k in b.params.cation_set
+                    if i != k)
+            elif (i in b.params.anion_set and j in b.params.cation_set):
+                return sum(Y[k] * _G_appr(
+                    b, pobj, (j+", "+i), (j+", "+k), b.temperature)
+                    for k in b.params.anion_set
+                    if i != k)
+            else:
+                raise BurntToast(
+                    "{} eNRTL model encountered unexpected component pair {}."
+                    .format(b.name, (i, j)))
+        b.add_component(pname+"_G",
+                        Expression(b.params.true_species_set,
+                                   b.params.true_species_set,
+                                   rule=rule_G_expr,
+                                   doc="Local interaction G term"))
+
```

Comment:
I may have missed this, but is the long-range interaction term also accounted for? If it isn't accounted for here, is there some reasoning for excluding the long-range G term?

Reply:
Yes - the long range contribution is in the next PR. And actually, so is the local contribution - this only sets up the parameters required.